src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,27 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Transaction transaction = session.beginTransaction();
+            try {
+                session.persist(movie);
+                transaction.commit();
+            } catch (Exception e) {
+                if (transaction != null) {
+                    transaction.rollback();
+                }
+                throw new DataProcessingException(
+                        "Can not to add movie with title: " + movie.getTitle());
+            }
+        }
+        return movie;
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        Session session = factory.openSession();
+        Movie movie = session.get(Movie.class, id);
+        session.close();
+        return Optional.ofNullable(movie);
     }
```

Comment:
Please use try with resources here. Also, you can use this: Optional.ofNullable(session.get(Movie.class, id))